pkg/bpf/detect.go

```diff
@@ -70,6 +72,47 @@ func HasKprobeMulti() bool {
 	return kprobeMulti.detected
 }
 
+func detectUprobeMulti() bool {
+	prog, err := ebpf.NewProgram(&ebpf.ProgramSpec{
+		Name: "probe_upm_link",
+		Type: ebpf.Kprobe,
+		Instructions: asm.Instructions{
+			asm.Mov.Imm(asm.R0, 0),
+			asm.Return(),
+		},
+		AttachType: ebpf.AttachTraceUprobeMulti,
+		License:    "MIT",
+	})
+	if errors.Is(err, unix.E2BIG) {
+		// Kernel doesn't support AttachType field.
+		return false
+	}
+	if err != nil {
+		return false
+	}
+	defer prog.Close()
+
+	ex, err := link.OpenExecutable("/proc/self/exe")
+	if err != nil {
+		return false
+	}
+
+	// need cilium/ebp fix, can't pass just addresses without symbol
```

Comment:
Is there an issue for this on cilium/ebpf? Would be nice to have a reference to check when this is fixed and update the code.

nit: typo -> cilium/ebp

Reply:
ah, the comment is no longer valid, it got fixed as preparation for uprobe multi support in ebpf/cilium lib
now it is:

```
func (ex *Executable) addresses(symbols []string, addresses, offsets []uint64) ([]uint64, error) {
        n := len(symbols)
        if n == 0 {
                n = len(addresses)
        }
```
will remove the comment and fix the typo, thanks